### ISSUE TITLE
EBADF bad file descriptor

### ISSUE THREAD

**Issue by freddi301** [stackblitz/webcontainer-core#1075]
### Describe the bug

It seems that not all file system primitives are implemented.

I get this error:

```
[Error [EBADF]: EBADF: bad file descriptor, read] {
  syscall: 'read',
  errno: -9,
  code: 'EBADF'
}
```

### Link to the blitz that caused the error

https://stackblitz.com/edit/node-nxhv96

### Steps to reproduce

1. `node idris2.node.js`

### Expected behavior

Do not crash

### Parity with Local

- [X] I have [run the project in my local machine](https://github.com/stackblitz/webcontainer-core/blob/main/repro.md#try-locally) and I could not reproduce the issue.

### Platform

```
Browser name  = Chrome
Full version  = 113.0.0.0
Major version = 113
navigator.appName = Netscape
navigator.userAgent = Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/113.0.0.0 Safari/537.36
performance.memory = {
  "totalJSHeapSize": 81215363,
  "usedJSHeapSize": 75249755,
  "jsHeapSizeLimit": 4294705152
}
Hash = 9c8944bc
```

**Comment by SamVerschueren** [comment]
@freddi301 Is it possible you shared the wrong link? The link just opens a default Node.js project, I don't see a file called `idris2.node.js`.

**Comment by freddi301** [comment]
I synced the container with this github repository https://github.com/freddi301/idris2-node-dist

Maybe that'why the contents are not visible?

**Comment by SamVerschueren** [comment]
Yeah not sure why the contents are gone. Can you update your opening post with this URL so it's easier for someone chiming in? https://stackblitz.com/~/github.com/freddi301/idris2-node-dist

I ran the file and indeed, running into the EBADF. I'll create a new internal issue to track this.

**Comment by NullVoxPopuli** [comment]
I'm getting EBADF on both of these blitz (both from GH):
- https://stackblitz.com/github/nullVoxPopuli/polaris-starter/tree/monorepo
- https://stackblitz.com/github/nullVoxPopuli/polaris-starter/tree/main

**Comment by SuperOleg39** [comment]
Hello!

Get the same error - https://stackblitz.com/edit/node-fk1xaf?file=tramvai.json

Few years ago catch the same error - https://github.com/stackblitz/webcontainer-core/issues/77#issuecomment-913517115

**Comment by SuperOleg39** [comment]
We use fastify, async-local-storage, can share any details about our framework internals.
